src/coreservices.h

```diff
@@ -153,6 +153,7 @@ class CoreServices : public QObject {
 
     Timer m_runtime_timer;
     const CmdlineArgs& m_cmdlineArgs;
+    bool m_isInitialized;
```

Comment:
We actually need 3 states for proper state tracking and assertions:
- Created
- Initialized
- Finalized

Reply:
> We actually need 3 states for proper state tracking and assertions:
> 
> * Created

The existence of the CoreServices object already implies that.

> * Initialized

This is tracked by this boolean.

> * Finalized

Shutting down without destroying `CoreServices` is not possible, so tracking this state is not necessary IMHO.